pkg/agent/baker.go

```diff
@@ -448,35 +448,35 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
 		},
 		"GetMasterOSImageOffer": func() string {
 			cloudSpecConfig := cs.GetCloudSpecConfig()
-			return fmt.Sprintf("\"%s\"", cloudSpecConfig.OSImageConfig[cs.Properties.MasterProfile.Distro].ImageOffer)
+			return fmt.Sprintf("\"%s\"", cloudSpecConfig.OSImageConfig[api.Distro(cs.Properties.MasterProfile.Distro)].ImageOffer)
 		},
 		"GetMasterOSImagePublisher": func() string {
 			cloudSpecConfig := cs.GetCloudSpecConfig()
-			return fmt.Sprintf("\"%s\"", cloudSpecConfig.OSImageConfig[cs.Properties.MasterProfile.Distro].ImagePublisher)
+			return fmt.Sprintf("\"%s\"", cloudSpecConfig.OSImageConfig[api.Distro(cs.Properties.MasterProfile.Distro)].ImagePublisher)
 		},
 		"GetMasterOSImageSKU": func() string {
 			cloudSpecConfig := cs.GetCloudSpecConfig()
-			return fmt.Sprintf("\"%s\"", cloudSpecConfig.OSImageConfig[cs.Properties.MasterProfile.Distro].ImageSku)
+			return fmt.Sprintf("\"%s\"", cloudSpecConfig.OSImageConfig[api.Distro(cs.Properties.MasterProfile.Distro)].ImageSku)
 		},
 		"GetMasterOSImageVersion": func() string {
 			cloudSpecConfig := cs.GetCloudSpecConfig()
-			return fmt.Sprintf("\"%s\"", cloudSpecConfig.OSImageConfig[cs.Properties.MasterProfile.Distro].ImageVersion)
+			return fmt.Sprintf("\"%s\"", cloudSpecConfig.OSImageConfig[api.Distro(cs.Properties.MasterProfile.Distro)].ImageVersion)
 		},
```

Comment:
we can remove all of these master os related functions, if you want.

Reply:
This code is removed in https://github.com/Azure/AgentBaker/pull/240